controller/volume_attachment_controller.go

```diff
@@ -342,13 +352,64 @@ func (vac *VolumeAttachmentController) handleVolumeMigrationConfirmation(va *lon
 			break
 		}
 	}
-	if !hasCSIAttachmentTicketRequestingPrevNode {
+	migratingEngineSnapSynced, err := vac.checkMigratingEngineSyncSnapshots(vol)
+	if err != nil {
+		vac.logger.WithError(err).Warn("failed to check migrating engine snapshot status")
+		return
+	}
+	if !hasCSIAttachmentTicketRequestingPrevNode && migratingEngineSnapSynced {
 		// TODO: Do we need check if the volume is available on the currentMigrationIDNode?
 		vol.Spec.NodeID = vol.Status.CurrentMigrationNodeID
 		vol.Spec.MigrationNodeID = ""
 	}
 }
 
+func (vac *VolumeAttachmentController) checkMigratingEngineSyncSnapshots(vol *longhorn.Volume) (bool, error) {
+	engines, err := vac.ds.ListVolumeEngines(vol.Name)
+	if err != nil {
+		return false, err
+	}
+
+	var migratingEngine *longhorn.Engine
+	var oldEngine *longhorn.Engine
+	for _, e := range engines {
+		if e.Spec.Active {
+			oldEngine = e
+			continue
+		}
+		if e.Spec.NodeID == vol.Spec.MigrationNodeID {
+			migratingEngine = e
+		}
+	}
+
+	if oldEngine == nil {
+		return false, fmt.Errorf("failed to find the active engine for volume %v", vol.Name)
+	}
+
+	if migratingEngine == nil {
+		return false, fmt.Errorf("failed to find the migrating engine for volume %v", vol.Name)
+	}
+
+	if !reflect.DeepEqual(oldEngine.Status.Snapshots, migratingEngine.Status.Snapshots) {
+		vac.logger.Debugf("Volume migration is in progress for synchronizing snapshots from old snapshot list %v to new %v", oldEngine.Status.Snapshots, migratingEngine.Status.Snapshots)
```

Comment:
This log would be quite tedious if the number of snapshots is a lot. Suggest log like this. `Volume migration (%v) is in progress for synchronizing snapshots` with volume name should be good enough.

Reply:
Modified. Thanks.